src/main/java/org/springframework/integration/aws/config/xml/SqsOutboundChannelAdapterParser.java

```diff
@@ -34,13 +34,37 @@
  */
 public class SqsOutboundChannelAdapterParser extends AbstractOutboundChannelAdapterParser {
 
+    public static final String QUEUE_MESSAGING_TEMPLATE_REF = "queue-messaging-template";
+
+    private boolean hasAttribute(Element element, String attribute) {
+        return StringUtils.hasText(element.getAttribute(attribute));
+    }
+
 	@Override
 	protected AbstractBeanDefinition parseConsumer(Element element, ParserContext parserContext) {
 		BeanDefinitionBuilder builder = BeanDefinitionBuilder.genericBeanDefinition(SqsMessageHandler.class);
-		builder.addConstructorArgReference(element.getAttribute(AmazonWSParserUtils.SQS_REF));
-		if (StringUtils.hasText(element.getAttribute(AmazonWSParserUtils.RESOURCE_ID_RESOLVER_REF))) {
+
+        if (hasAttribute(element, QUEUE_MESSAGING_TEMPLATE_REF) &&
```

Comment:
Looks like you have `4 whitespaces` for indents not tabs.
I can fix that on merge, but keep in mind for the future contributions.